fandogh_cli/image_commands.py

```diff
@@ -111,7 +111,7 @@ def publish(version, detach):
             return
     workspace = Workspace()
 
-    if not workspace.has_docker_file:
+    if not workspace.docker_file_name:
```

Comment:
let's have 2 separated changeset. 
one for dockerignore bug and the other for dockerignore improvement.

Regarding the bug it seems to me we only need to change the `has_docker_ignore` logic to sth like:
`self.has_docker_file = 'dockerfile' in map(str.lower, files)`

do you have time to test it? I believe it's enough. 
once we merge it we can improve the docker ignore part.